java/arcs/core/allocator/Allocator.kt

```diff
@@ -55,6 +55,16 @@ class Allocator(val hostRegistry: HostRegistry) {
         return arcId
     }
 
+    /**
+     * Stop an Arc given its [ArcId].
+     */
+    suspend fun stopArc(arcId: ArcId ) {
+        val partitions = readPartitionMap(arcId) ?: return
+        // Store computed partitions for later
+        writePartitionMap(arcId, partitions)
```

Comment:
why read and then immediately write? I think you're want to move this writePartitionMap() call after the stopPlan() call otherwise you're not writing the changed state (ArcState.Stopped)

Reply:
This line should be removed.